pkg/resource/cluster/pre_build_request.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cluster
+
+import (
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	"github.com/pkg/errors"
+	"strconv"
+
+	"github.com/aws-controllers-k8s/runtime/pkg/requeue"
+)
+
+// validateClusterNeedsUpdate this function's purpose is to requeue if the resource is currently unavailable and
+// to validate if resource update is required.
+func (rm *resourceManager) validateClusterNeedsUpdate(
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	// requeue if necessary
+	latestStatus := latest.ko.Status.Status
+	if latestStatus == nil || *latestStatus != available {
+		return nil, requeue.NeededAfter(
+			errors.New("cluster cannot be updated as its status is not 'available'"),
```

Comment:
Please use `fmt.Errorf`:
```suggestion
			fmt.Errorf("cluster cannot be updated as its status is not '%s'", StatusAvailable),
```